### PR TITLE
Deal with function in returned objects

### PR DESCRIPTION
**Pull request by asmab89**

### CODE REVIEW

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLActionBuilder.kt [hunk 1]
```diff
@@ -732,7 +734,7 @@ object GraphQLActionBuilder {
             methodType.equals(GqlConst.QUERY_TYPE, true) -> GQMethodType.QUERY
             methodType.equals(GqlConst.MUTATION, true) -> GQMethodType.MUTATION
             else -> {
-                //TODO log warn
+                log.warn("Entry point is not found.")
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLActionBuilder.kt:737]: when issuing warning, be bit more descriptive

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLActionBuilder.kt [hunk 2]
```diff
@@ -897,30 +901,15 @@ object GraphQLActionBuilder {
...
-                && !(gene is ArrayGene<*> && gene.template is EnumGene<*>)
-                && !(gene is ArrayGene<*> && gene.template is OptionalGene && gene.template.gene is EnumGene<*>)
-                && !(gene is OptionalGene && gene.gene is ArrayGene<*> && gene.gene.template is EnumGene<*>)
-                && !(gene is EnumGene<*>)
-                && !(gene is OptionalGene && gene.gene is EnumGene<*>)
-
-            ) {
-                params.add(GQReturnParam(methodName, gene))
-            }
+            if (isReturnPrimitive(gene)) params.add(GQReturnParam(methodName, gene))
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLActionBuilder.kt:909]: is the comment above correct? looks like you are adding it when a primitive

File: core/src/main/kotlin/org/evomaster/core/search/gene/GeneUtils.kt [hunk 3]
```diff
@@ -485,19 +494,34 @@ object GeneUtils {
...
-    fun shouldApplyBooleanSelection(gene: Gene) =
-            (gene is OptionalGene && gene.gene is ObjectGene)
-                    || gene is ObjectGene
-                    || (gene is ArrayGene<*> && gene.template is ObjectGene)
-                    || (gene is ArrayGene<*> && gene.template is OptionalGene && gene.template.gene is ObjectGene)
-                    || (gene is OptionalGene && gene.gene is ArrayGene<*> && gene.gene.template is OptionalGene && gene.gene.template.gene is ObjectGene)
-                    || (gene is OptionalGene && gene.gene is ArrayGene<*> && gene.gene.template is ObjectGene)
+    private fun isLastElementInTupleOptional(lastElement: Gene) {
+        if (lastElement is OptionalGene) {
+            lastElement.isActive = true
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/search/gene/GeneUtils.kt:508]: this is not valid.
you have a method whose name starts with `is`, which implies check for boolean, which this method does not return.
furthermore, this method has a side effect on the input argument `lastElement`.
name of function must be changed

File: core/src/test/kotlin/org/evomaster/core/search/gene/GeneUtilsTest.kt [hunk 4]
```diff
@@ -350,4 +353,64 @@ internal class GeneUtilsTest {
         assertEquals("{foo{...onA{a1}b1,b2}}", res)//with the name foo and without "...on" for the object B
     }
 
+    /*
+    The tests below are temporary, to remove
+     */
```
Comment (arcuri82) [inline core/src/test/kotlin/org/evomaster/core/search/gene/GeneUtilsTest.kt:358]: so, shouldn't you remove them? :)

**Review by arcuri82 — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLActionBuilder.kt (hunk 1 above)
Comment (asmab89) [reply core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLActionBuilder.kt]: done

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLActionBuilder.kt (hunk 2 above)
Comment (asmab89) [reply core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLActionBuilder.kt]: done

File: core/src/main/kotlin/org/evomaster/core/search/gene/GeneUtils.kt (hunk 3 above)
Comment (asmab89) [reply core/src/main/kotlin/org/evomaster/core/search/gene/GeneUtils.kt:508]: done

File: core/src/test/kotlin/org/evomaster/core/search/gene/GeneUtilsTest.kt (hunk 4 above)
Comment (asmab89) [reply core/src/test/kotlin/org/evomaster/core/search/gene/GeneUtilsTest.kt]: done

**Pull request merged**